packages/flutter_tools/lib/src/globals.dart

```diff
@@ -308,3 +309,22 @@ final RegExp kVMServiceMessageRegExp = RegExp(r'The Dart VM service is listening
 // The official tool no longer allows non-null safe builds. This can be
 // overridden in other clients.
 NonNullSafeBuilds get nonNullSafeBuilds => context.get<NonNullSafeBuilds>() ?? NonNullSafeBuilds.notAllowed;
+
+bool _searchedForJava = false;
+Java? _javaInstance;
+
+Java? get java => context.get<Java>() ?? (() {
```

Comment:
why do we need a fallback here? Should we not instead construct this in context_runner.dart?

Reply:
I forgot that context_runner already provides lazy loading, so I removed this.